### PR TITLE
OH2-400 | OH2-403 Refactoring ResponseEntity

### PR DESCRIPTION
**Pull request by SilverD3**
See OH2-400

### Tasks

- [x] `org/isf/accounting/rest/BillController.java`
- [x] `org/isf/admission/rest/AdmissionController.java`
- [x] `org/isf/admtype/rest/AdmissionTypeController.java`
- [x] `org/isf/agetype/rest/AgeTypeController.java`
- [x] `org/isf/disctype/rest/DischargeTypeController.java`
- [x] `org/isf/disease/rest/DiseaseController.java`
- [x] `org/isf/distype/rest/DiseaseTypeController.java`
- [x] `org/isf/dlvrrestype/rest/DeliveryResultTypeController.java`
- [x] `org/isf/dlvrtype/rest/DeliveryTypeController.java`
- [x] `org/isf/exam/rest/ExamController.java`
- [x] `org/isf/exam/rest/ExamRowController.java`
- [x] `org/isf/examination/rest/ExaminationController.java`
- [x] `org/isf/exatype/rest/ExamTypeController.java`
- [x] `org/isf/hospital/rest/HospitalController.java`
- [x] `org/isf/lab/rest/LaboratoryController.java`
- [x] `org/isf/login/rest/LoginController.java`
- [x] `org/isf/malnutrition/rest/MalnutritionController.java`
- [x] `org/isf/medical/rest/MedicalController.java`
- [x] `org/isf/medicalstock/rest/MedicalStockMovementController.java`
- [x] `org/isf/medicalstock/rest/StockMovementController.java`
- [x] `org/isf/medicalstockward/rest/MedicalStockWardController.java`
- [x] `org/isf/medstockmovtype/rest/MedStockMovementTypeController.java`
- [x] `org/isf/medtype/rest/MedicalTypeController.java`
- [x] `org/isf/opd/rest/OpdController.java`
- [x] `org/isf/operation/rest/OperationController.java`
- [x] `org/isf/opetype/rest/OperationTypeController.java`
- [x] `org/isf/patconsensus/rest/PatientConsensusController.java`
- [x] `org/isf/patient/rest/PatientController.java`
- [x] `org/isf/patvac/rest/PatVacController.java`
- [x] `org/isf/permissions/rest/PermissionController.java`
- [x] `org/isf/pregtreattype/rest/PregnantTreatmentTypeController.java`
- [x] `org/isf/priceslist/rest/PriceListController.java`
- [x] `org/isf/pricesothers/rest/PricesOthersController.java`
- [x] `org/isf/sms/rest/SmsController.java`
- [x] `org/isf/stats/rest/ReportsController.java`
- [x] `org/isf/supplier/rest/SupplierController.java`
- [x] `org/isf/therapy/rest/TherapyController.java`
- [x] `org/isf/usergroups/rest/UserGroupController.java`
- [x] `org/isf/users/rest/UserController.java`
- [x] `org/isf/usersettings/rest/UserSettingController.java`
- [x] `org/isf/vaccine/rest/VaccineController.java`
- [x] `org/isf/vactype/rest/VaccineTypeController.java`
- [x] `org/isf/visits/rest/VisitsController.java`
- [x] `org/isf/ward/rest/WardController.java`

### CODE REVIEW

File: src/main/java/org/isf/accounting/rest/BillController.java [hunk 1]
```diff
@@ -199,13 +203,13 @@ public ResponseEntity<FullBillDTO> updateBill(@PathVariable Integer id, @Request
 	 * @param dateTo the high date range endpoint, inclusive.
 	 * @param code the patient code, which can be set or not.
 	 * @return a list of retrieved {@link Bill}s or {@code null} if an error occurred.
-	 * @throws OHServiceException
+	 * @throws OHServiceException When failed to get bills
 	 */
-	@GetMapping(value = "/bills", produces = MediaType.APPLICATION_JSON_VALUE)
+	@GetMapping("/bills")
 	public ResponseEntity<List<BillDTO>> searchBills(
```
Comment (SteveGT96) [inline src/main/java/org/isf/accounting/rest/BillController.java]:
```suggestion
	@GetMapping("/bills")
	public List<BillDTO> searchBills(
```

**Review by SteveGT96 — CHANGES_REQUESTED** [review]
It would be better to remoive also response entity on endoint like this one. Instead of returning NoConent when the list us empty or null, we should return an empty list with status 200.

**Comment by mwithi** [comment]
Please resolve conflicts

**Comment by mwithi** [comment]
@SilverD3 please merge develop and update again the spec file, thanks

**Comment by SilverD3** [comment]
@mwithi up-to-date

**Comment by mwithi** [comment]
> @mwithi up-to-date

Thanks. Are all these breaking changes meant to be?

https://github.com/informatici/openhospital-api/actions/runs/11572320351/job/32212045163?pr=482#step:19:19

**Comment by SilverD3** [comment]
Yes. I will align the UI

**Comment by mwithi** [comment]
> Yes. I will align the UI

Ok, please, start doing it and create a PR on UI with same branch name, thanks!

**Comment by SilverD3** [comment]
@mwithi paired PR on the UI already created

**Comment by mwithi** [comment]
> git pull https://github.com/SilverD3/openhospital-api.git OH2-400_responseentity_refactoring

best way to test is to do random tests on any endpoint with the new UI, right?

**Comment by mwithi** [comment]
> @mwithi paired PR on the UI already created

@SilverD3 best way to test is to do random tests on any endpoint with the new UI, right?

**Comment by SilverD3** [comment]
> @SilverD3 best way to test is to do random tests on any endpoint with the new UI, right?

Yes, just to ensure that nothing is broken with the removal of `ResponseEntity`. Anyway, integration tests on the API are passing, so this PR is mostly a formality :)

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**